app/controllers/exercises_controller.rb

```diff
@@ -30,7 +30,9 @@ def start!
   end
 
   def set_default_content
+    @file_content = @exercise.files_content_for(current_user).except('content')
     @current_content = @exercise.current_content_for(current_user)
+    @files = @file_content.keys
```

Comment:
Also, instead of  `file_contents_for` returning a hash - which leads to complex, imperative code - why not returning an array of structs with both a name a content? That way you don't need the `@files` attribute: you can just iterate over the files and get their names and content on demand.

Reply:
I disagree. Tomorrow we will discuss this